### ISSUE TITLE
Compare performance of C and C# implementations

### ISSUE THREAD

**Issue by Konard** [Konard/LinksPlatform#30]

**Comment by werkzeuge** [comment]
Если четыре ядра процессора (или четыре потока) - будет работать с четырьмя деревьями (структурами) одновременно?

**Comment by Konard** [comment]
> Если четыре ядра процессора (или четыре потока) - будет работать с четырьмя деревьями (структурами) одновременно?
Можно пробовать, достаточно запустить Links 4 раза и дать пути к 4-ём разным файлам базы данных, тогда повреждений данных не будет. Но внутри одной базы данных многопоточность либо не гарантируется (могут быть повреждения), либо ставится полная блокировка на всю базу данных при записи. Решение этой проблемы пока в будущем.

**Comment by Konard** [comment]
> Если четыре ядра процессора (или четыре потока) - будет работать с четырьмя деревьями (структурами) одновременно?

Нет, если доступны только 2 канала памяти, 3-й и 4-й процессор будут просто ждать пока контроллер памяти освободится.
